File: Java/src/Silver3/B11726.java
package Silver3;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class B11726 {
    public static void main(String[] args) throws IOException {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] cnt = new int[n + 1];
        cnt[1] = 1;
        if(n > 1) {
        	cnt[2] = 2;
        }
        for (int i = 3; i <= n; i++) {
            cnt[i] = (cnt[i - 1] + cnt[i - 2]) % 10007;
        }
        System.out.println(cnt[n]);
    }
}

File: Silver3/B11726.py
# 2×n 타일링
import sys

n = int(sys.stdin.readline())
cnt = [0] * (n+1) # 핵심

cnt[1] = 1
if n > 1:
    cnt[2] = 2
for i in range(3, n+1):
    cnt[i] = (cnt[i-1] + cnt[i-2]) % 10007

print(cnt[n])